External/Plugins/HaXeContext/Linters/DiagnosticsLinter.cs

```diff
@@ -33,44 +34,51 @@ public void LintAsync(string[] files, LintCallback callback)
-                    if (results == null)
+                    var hc = context.GetHaxeComplete(document.SciControl, exprStub, true, HaxeCompilerService.DIAGNOSTICS);
+                    hc.GetDiagnostics((complete, results, status) =>
                     {
-                        callback(null);
-                        return;
-                    }
-
-                    foreach (var res in results)
-                    {
-                        var line = res.Range.LineStart + 1;
-                        var firstChar = document.SciControl.PositionFromLine(line) + res.Range.CharacterStart;
-                        var lastChar = document.SciControl.PositionFromLine(res.Range.LineEnd + 1) + res.Range.CharacterEnd;
-                        var result = new LintingResult
+                        var list = new List<LintingResult>();
+                        if (results == null)
                         {
-                            File = res.Range.Path,
-                            FirstChar = res.Range.CharacterStart,
-                            Length = lastChar - firstChar,
-                            Line = line,
-                            Severity = conversionDict[res.Severity]
-                        };
+                            callback(null);
+                            return;
+                        }
 
-                        switch (res.Kind)
+                        foreach (var res in results)
                         {
-                            case HaxeDiagnosticsKind.UNUSEDIMPORT:
-                                result.Description = "Unused import";
-                                break;
-                            case HaxeDiagnosticsKind.UNUSEDVAR:
-                                result.Description = "Unused variable";
-                                break;
+                            var line = res.Range.LineStart + 1;
+                            var firstChar = document.SciControl.PositionFromLine(line) + res.Range.CharacterStart;
+                            var lastChar = document.SciControl.PositionFromLine(res.Range.LineEnd + 1) + res.Range.CharacterEnd;
+                            var result = new LintingResult
+                            {
+                                File = res.Range.Path,
+                                FirstChar = res.Range.CharacterStart,
+                                Length = lastChar - firstChar,
+                                Line = line,
+                                Severity = conversionDict[res.Severity]
+                            };
+
+                            switch (res.Kind)
+                            {
+                                case HaxeDiagnosticsKind.UNUSEDIMPORT:
+                                    result.Description = "Unused import";
+                                    break;
+                                case HaxeDiagnosticsKind.UNUSEDVAR:
+                                    result.Description = "Unused variable";
```

Comment:
I guess this should be localized. And maybe we should create some `Kind`, `ResultId` or similar property for `LintingResult` to be able to maybe link it in the future with some other stuff? Although with proper implementation case I guess adding this already will surely mean it will be changed in the future at the first time we will really need it.